common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -143,7 +145,24 @@ public static boolean execICU(final UTF8String l, final UTF8String r,
    * Collation-aware regexp expressions.
    */
 
-  // TODO: Add more collation-aware regexp expressions.
+  public static boolean supportsLowercaseRegex(final int collationId) {
```

Comment:
Should we add thorough unit tests for these functions in `CollationSupportSuite`?

Reply:
there is no behaviour to test in CollationSupportSuite, all collation awareness here is embedded directly in regexpExpressions instead

this one is gonna have to be a bit different than StringExpressions (in terms of testing)